src/gluonts/model/predictor.py

```diff
@@ -134,6 +155,79 @@ def from_inputs(cls, train_iter, **params):
         params = {**auto_params, **params}
         return cls.from_hyperparameters(**params)
 
+    def get_backtest_input(
+        self,
+        test_data: TestData,
+        forecasts: Iterable[Union[Forecast, ForecastBatch]],
+    ) -> Iterator[Dict[str, np.ndarray]]:
+        for input, label, forecast in zip(
+            test_data.input, test_data.label, forecasts
+        ):
+            input_target = input["target"]
+            label_target = label["target"]
+
+            non_forecast_data = {
+                "label": label_target,
+                "seasonal_error": seasonal_error(
+                    input_target,
+                    seasonality=get_seasonality(
+                        freq=forecast.start_date.freqstr
+                    ),
+                ),
+            }
+
+            batching_used = isinstance(forecast, ForecastBatch)
+            if batching_used:
+                forecast_data = forecast
+            else:
+                non_forecast_data = {
+                    key: np.expand_dims(value, axis=0)
+                    for key, value in non_forecast_data.items()
+                }
+
+                if isinstance(forecast, SampleForecast):
+                    forecast_data = SampleForecastBatch.from_forecast(forecast)
+                elif isinstance(forecast, QuantileForecast):
+                    forecast_data = QuantileForecastBatch.from_forecast(
+                        forecast
+                    )
+
+            joint_data = ChainMap(non_forecast_data, forecast_data)
+
+            yield joint_data
+
+    def backtest(
+        self,
+        test_data: TestData,
+        metrics: Collection[Metric],
```

Comment:
I wonder if we would want to swap the order of the two arguments.

Reply:
This way it is somewhat consistent with `get_backtest_input` as `test_data` is the first argument there as well. But that's not much of a reason to keep it like it is.

I agree that swapping them here emphasizes that the most interesting part of this are the metrics that we want to evaluate.